File: desktop-tauri/src-tauri/python/context_profiles.py
"""Static context profile registry for the desktop compute-node bridge."""
from __future__ import annotations

from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True)
class ContextProfile:
    profile_id: str
    display_label: str
    total_context_tokens: int
    default_output_reservation_tokens: int
    enabled: bool = True


DEFAULT_CONTEXT_TIER = "8k-fast"
CONTEXT_PROFILES = {
    "8k-fast": ContextProfile("8k-fast", "8K Fast", 8192, 512),
    "64k-full": ContextProfile("64k-full", "64K Full", 65536, 1024),
}


def resolve_context_profile(profile_id: str) -> Optional[ContextProfile]:
    profile = CONTEXT_PROFILES.get((profile_id or "").strip())
    if profile is None or not profile.enabled:
        return None
    return profile


def normalize_context_tier(profile_id: str) -> str:
    profile = resolve_context_profile(profile_id)
    return profile.profile_id if profile else DEFAULT_CONTEXT_TIER


def require_context_profile(profile_id: str) -> ContextProfile:
    profile = resolve_context_profile(profile_id)
    if profile is None:
        raise ValueError(f"unknown context profile: {profile_id}")
    return profile

File: desktop-tauri/src-tauri/src/context_profiles.rs
use serde::{Deserialize, Serialize};

pub const DEFAULT_CONTEXT_TIER_ID: &str = "8k-fast";

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub struct ContextProfile {
    pub id: &'static str,
    pub display_label: &'static str,
    pub total_context_tokens: u32,
    pub default_output_token_reservation: u32,
    pub enabled: bool,
}

pub const CONTEXT_PROFILES: &[ContextProfile] = &[
    ContextProfile {
        id: "8k-fast",
        display_label: "8K Fast",
        total_context_tokens: 8_192,
        default_output_token_reservation: 512,
        enabled: true,
    },
    ContextProfile {
        id: "64k-full",
        display_label: "64K Full",
        total_context_tokens: 65_536,
        default_output_token_reservation: 1_024,
        enabled: true,
    },
];

pub fn context_profile(id: &str) -> Option<&'static ContextProfile> {
    CONTEXT_PROFILES
        .iter()
        .find(|profile| profile.enabled && profile.id == id)
}

pub fn normalize_context_tier(id: &str) -> String {
    context_profile(id)
        .map(|profile| profile.id.to_string())
        .unwrap_or_else(|| DEFAULT_CONTEXT_TIER_ID.to_string())
}

pub fn default_context_tier() -> String {
    DEFAULT_CONTEXT_TIER_ID.to_string()
}
